src/details/ArborX_DetailsDistributor.hpp

```diff
@@ -173,18 +167,13 @@ class Distributor
         Kokkos::ViewAllocateWithoutInitializing("destination_buffer"),
         exports.size());
 
-    Kokkos::View<int *, typename View::traits::device_type> permute_mirror(
-        Kokkos::ViewAllocateWithoutInitializing("permute_device_mirror"),
-        _permute.size());
-    Kokkos::deep_copy(permute_mirror, _permute);
-
     Kokkos::parallel_for("copy_destinations_permuted",
                          Kokkos::RangePolicy<ExecutionSpace>(
                              0, _dest_offsets.back() * num_packets),
                          KOKKOS_LAMBDA(int const k) {
                            int const i = k / num_packets;
                            int const j = k % num_packets;
-                           dest_buffer(num_packets * permute_mirror[i] + j) =
+                           dest_buffer(num_packets * _permute[i] + j) =
```

Comment:
Wait how does that work if we are not using CUDA-aware MPI and `_permute` lives on the host?

Reply:
If we are not using CUDA-aware MPI, the function only deals with host space. There is an assert for that a few lines above.

Reply:
Right but the assertion you'd need is that `_permute` is accessible from the execution space.  Am I missing something?

Reply:
If we don't have `CUDA-aware MPI` we always copy all the data to the host before calling this function. The execution space is given by the input view. Hence, it is a host execution space.

Reply:
OK.  Please assert it, it will improve readability.

Reply:
CI says illegal memory accesses in the CUDA-aware build.  `dest_buffer` lives on the host :)

Reply:
OK. This is strange I can assert that the types for `_permute` and `permute_mirror` are the same, but the code crahes if I use `_permute` instead of `permute_mirror`.

Reply:
So copy constructing a new variable and using that instead of `_permute` also works (which is expected since this is what `create_mirror_view_and_copy` is supposed to do here anyway).

Reply:
I missed that this requires to capture the enclosing object by copy which we can't (easily) do in `Kokkos` before C++17. Hence, it is necessary to create a local copy.